From: Kaitlin Rupert <kaitlin.rupert@intel.com>
Date: Thu, 20 May 2010
Subject: The 4.7.0 pre-release version of QT does not have toMSecsSinceEpoch()

Once the version of Qt is upgraded, this patch can be removed.


--- a/src/corelib/i18n/mcalendar.cpp
+++ b/src/corelib/i18n/mcalendar.cpp
@@ -249,14 +249,14 @@ void MCalendar::setDateTime(QDateTime da
 
     // We cannot use QDateTime::toTime_t because this
     // works only for dates after 1970-01-01T00:00:00.000.
-#if QT_VERSION >= 0x040700
-    UDate icuDate = date.toMSecsSinceEpoch();
-#else
+//#if QT_VERSION >= 0x040700
+//    UDate icuDate = date.toMSecsSinceEpoch();
+//#else
     // Qt < 4.7 lacks QDateTime::toMSecsSinceEpoch(), we need to emulate it:
     int days = QDate(1970, 1, 1).daysTo(date.date());
     qint64 msecs = qint64(QTime().secsTo(date.time())) * 1000;
     UDate icuDate = (qint64(days) * MSECS_PER_DAY) + msecs;
-#endif
+//#endif
 
     if (originalTimeSpec == Qt::LocalTime) {
         // convert from local time to UTC
@@ -292,9 +292,9 @@ QDateTime MCalendar::qDateTime(Qt::TimeS
     }
     // We cannot use QDateTime::setTime_t because this
     // works only for dates after 1970-01-01T00:00:00.000.
-#if QT_VERSION >= 0x040700
-    time.setMSecsSinceEpoch(qint64(icuDate));
-#else
+//#if QT_VERSION >= 0x040700
+//    time.setMSecsSinceEpoch(qint64(icuDate));
+//#else
     // Qt < 4.7 lacks QDateTime::setMSecsSinceEpoch(), we need to emulate it.
     qint64 msecs = qint64(icuDate);
     int ddays = msecs / MSECS_PER_DAY;
@@ -306,7 +306,7 @@ QDateTime MCalendar::qDateTime(Qt::TimeS
     }
     time.setDate(QDate(1970, 1, 1).addDays(ddays));
     time.setTime(QTime().addMSecs(msecs));
-#endif
+//#endif
     // note: we set time spec after time value so Qt will not any conversions
     // of its own to UTC. We might let Qt handle it but this might be more robust
     time.setTimeSpec(spec);
